app/models/solidus_stripe/payment_method.rb

```diff
@@ -4,40 +4,153 @@ module SolidusStripe
   class PaymentMethod < ::Spree::PaymentMethod
     preference :api_key, :string
     preference :publishable_key, :string
+    preference :webhook_endpoint_secret, :string, default: ''
 
     validates :available_to_admin, inclusion: { in: [false] }
 
-    def partial_name
-      "stripe"
-    end
+    concerning :Actions do
+      def actions
+        %w[capture void credit]
+      end
 
-    alias cart_partial_name partial_name
-    alias product_page_partial_name partial_name
+      # @param payment [Spree::Payment] the payment we want to know if can be captured
+      # @return [Boolean] true when the payment is in the pending or checkout states
+      def can_capture?(payment)
+        payment.pending? || payment.checkout?
+      end
+
+      # @param payment [Spree::Payment] the payment we want to know if can be voided
+      # @return [Boolean] true when the payment is not failed, voided or invalid
+      def can_void?(payment)
+        payment.can_void? && !payment.completed?
+      end
 
-    def payment_source_class
-      PaymentSource
+      # Indicates whether its possible to credit the payment.  Note that most
+      # gateways require that the payment be settled first which generally
+      # happens within 12-24 hours of the transaction.
+      #
+      # @param payment [Spree::Payment] the payment we want to know if can be credited
+      # @return [Boolean] true when the payment is completed and can be credited
+      def can_credit?(payment)
+        payment.completed? && payment.credit_allowed > 0
+      end
     end
 
-    def gateway_class
-      Gateway
+    concerning :Configuration do
+      def source_required?
+        false
+      end
+
+      def payment_source_class
+        PaymentSource
+      end
+
+      def gateway_class
+        Gateway
+      end
+
+      def partial_name
+        "stripe"
+      end
+
+      alias cart_partial_name partial_name
+      alias product_page_partial_name partial_name
+      alias risky_partial_name partial_name
     end
 
-    def create_profile(payment)
-      payment_intent = payment.source.payment_intent
+    concerning :Order do
+      def find_in_progress_payment_for(order)
+        payments = order.payments
+          .where.not(state: %w[completed invalid void]) # in_progress
```

Comment:
Why not `.where(state: "pending")`?

Reply:
Isn't `processing` another option?

Reply:
Yeah, it's the one used when the payment element has been rendered but not filled yet. Probably it won't be used at any point, but it makes sense adding it.

Reply:
I see. That would be a good candidate to move upstream as a scope 👍 But let's move forward now.